projects/storefrontlib/src/lib/my-account/order/order-details/order-details.component.spec.ts

```diff
@@ -5,12 +5,12 @@ import { By } from '@angular/platform-browser';
 import createSpy = jasmine.createSpy;
 
 import { OrderDetailsComponent } from '../order-details/order-details.component';
-import { RoutingService } from '@spartacus/core';
+import { RoutingService, Order } from '@spartacus/core';
 import { UserService } from '../../../user/facade/user.service';
 import { AuthService } from '../../../auth/facade/auth.service';
 import { CardModule } from '../../../ui/components/card/card.module';
 
-const mockOrder = {
+const mockOrder: Order = {
```

Comment:
Can we add types on lines: 61, 74, 76, 84, 85, 86, 90, 99,102 and 137?

Reply:
line 74: items: any[] is never used. So I don't know what type it should be.
line 84: setting mockAuthService to be of type AuthService causes errors, and fixing the errors results in having to create valid UserToken, ClientToken, and Store objects, which is rather verbose.
line 85: setting mockRoutingService to be of type RoutingService causes errors
line 86: setting mockUserService to be of type UserService causes errors
line 90: refers to variable declared on line 85, already covered
line 99: refers to variable declared on line 84, already covered
line 102: refers to variable declared on line 86, already covered

Reply:
line 74 - in that case remove all the `@Input()`s that are not used (see my comments below)
line 84, 85, 86, 90, 99, 102 - discussed offline, there's a way to bypass those TS errors

Reply:
Removing the Input decorators / values causes insane errors... Leaving them..

Reply:
Ok, if they are used, try to add proper types to them.
`PromotionResult`.